### ISSUE TITLE
Ranking with time spent in a tweet

### ISSUE THREAD

**Issue by Steve-whb** [twitter/the-algorithm#1749]
This is not about a problem I've encountered, but about a small suggestion for ranking features used. So every time I use Twitter, I noticed that I usually spend more time on a tweet if I find it more interesting. As I read your recommendation system blog post and I came across this paragraph: 
```
Ranking is achieved with a ~48M parameter neural network that is continuously trained on Tweet interactions to optimize for positive engagement (e.g. Likes, Retweets, and Replies).
```

It looks like you don't use time spent on a tweet as a ranking feature so I think it could be beneficial for the recommendation quality.

**Comment by rohansanghvi33** [comment]
To be honest, that can be flawed logic. Anyone can write **long tweets** or incomprehensible tweets, just to keep people on the tweet for a longer amount of time.

**Comment by Steve-whb** [comment]
yeah, that's a good point. But I think being just one of the features used instead of the only feature used probably can mitigate the issue. Also if a tweet is only long for the sake of being long or simply incomprehensible, then for me I probably will just scan through it real quick without spending much time reading through it.

**Comment by Steve-whb** [comment]
> To be honest, that can be flawed logic. Anyone can write **long tweets** or incomprehensible tweets, just to keep people on the tweet for a longer amount of time.

yeah, that's a good point. But I think being just one of the features used instead of the only feature used probably can mitigate the issue. Also if a tweet is only long for the sake of being long or simply incomprehensible, then at least for me I probably will just scan through it real quick without spending much time reading through it.

**Comment by tas-pro** [comment]
A flaw with ranking tweets with this method is you don’t know if the user is actually reading the tweet. The user could probably have left the tab open and left the computer or is focused on other tab.

**Comment by ShahrozS** [comment]
Instead of the time spent on the tweet only, we can use time spent on the thread or the user scrolling through the replies of that tweet (basically expanding the tweet) implying that he is interested in the discussion.

**Comment by placholder** [comment]
just skip the incomprehensible
its that ez
